python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -240,12 +207,105 @@ async def setup_calsys(self, sequence_name: str) -> None:
+        location of the linear stage for the laser projector focus in mm
+
+        """
+        # TODO (DM-44772): implement the actual function
+        return 10.0
+
+    async def change_laser_wavelength(
+        self,
+        wavelength: float | None,
+        use_projector: typing.Optional[bool] = True,
+    ) -> None:
+        """Change the TunableLaser wavelength setting
+
+        Parameters
+        ----------
+        wavelength : `float`
+            wavelength of the laser in nm
+        use_projector : `bool`
+            identifies if you are using the projector while
+            changing the wavelength
+        """
+        if wavelength is not None:
+            task_wavelength = self.rem.tunablelaser.cmd_changeWavelength.set_start(
+                wavelength=wavelength, timeout=self.long_long_timeout
+            )
+
+        if use_projector:
+            task_focus = self.linearstage_laser_focus.cmd_moveAbsolute.set_start(
+                distance=self.calculate_laser_focus_location(wavelength),
+                timeout=self.long_long_timeout,
             )
+            await asyncio.gather(task_wavelength, task_focus)
 
-    async def setup_laser(self, mode: LaserDetailedState) -> None:
+        else:
+            await task_wavelength
+
+    async def change_laser_optical_configuration(
+        self, optical_configuration: LaserOpticalConfiguration
+    ) -> None:
+        """Change the output of the laser.
+
+        Parameters
+        ----------
+        optical_configuration : LaserOpticalConfiguration
+        """
+        current_configuration = (
+            await self.rem.tunablelaser.evt_opticalConfiguration.aget().configuration
+        )
+        if current_configuration != optical_configuration:
+            self.log.debug(
+                f"Changing optical configuration from {current_configuration} to {optical_configuration}"
+            )
+            if optical_configuration in list(LaserOpticalConfiguration):
+                await self.rem.tunablelaser.cmd_setOpticalConfiguration.set_start(
+                    configuration=optical_configuration, timeout=self.long_timeout
+                )
+            elif LaserOpticalConfiguration[optical_configuration] in list(
+                LaserOpticalConfiguration
+            ):
+                await self.rem.tunablelaser.cmd_setOpticalConfiguration.set_start(
+                    configuration=LaserOpticalConfiguration[optical_configuration],
+                    timeout=self.long_timeout,
+                )
+            else:
+                raise RuntimeError(
+                    f"{optical_configuration} not an acceptable LaserOpticalConfiguration"
+                )
+
+        else:
+            self.log.debug("Laser Optical Configuration already in place.")
+
+    async def setup_laser(
+        self,
+        mode: LaserDetailedState,
+        wavelength: typing.Optional[float] = 500.0,
+        optical_configuration: typing.Optional[
```

Comment:
Same thing here, no need for this to optional since you are only allowing one type.